File: Frontend/parth-traders/src/app/user/core/models/Order.ts
import { Customer, PaymentType } from "./Customer";
import { OrderDetail } from "./OrderDetail";

export class Order {
    transactionId: string;
    userName: string;
    orderDetails: OrderDetail[];
    paymentType: PaymentType;
    orderDate: Date;
    grandTotal: number;
    orderStatus: OrderStatus;

    constructor(customer: Customer, orderDetails: OrderDetail[]) {
        this.transactionId = '';
        this.userName = customer?.userName;
        this.orderDetails = orderDetails;
        this.grandTotal = this.orderDetails.reduce((acc, currentOrderDetail) => acc + currentOrderDetail.getTotalPrice(), 0);
        this.orderDate = new Date();
        this.paymentType = PaymentType.CashOnDelivery;
        this.orderStatus = OrderStatus.Pending;
    }

    getOrderDetails() {
        return this.orderDetails;
    }

    getTotalOrderPrice() {
        return this.orderDetails.reduce((acc, currentOrderDetail) => acc + currentOrderDetail.getTotalPrice(), 0);
    }
}

export enum OrderStatus {
    Pending,
    Approved,
    Processing,
    CanceledByCustomer,
    CanceledByOwner,
    Completed,
}

File: Backend/Parth Traders.Domain/Entity/User/Order.cs
using Parth_Traders.Domain.Enums;

namespace Parth_Traders.Domain.Entity.User
{
    public class Order
    {
        public long OrderId { get; set; }

        public string TransactionId { get; set; }

        public virtual Customer Customer { get; set; }

        public List<OrderDetail> OrderDetails { get; set; }

        public PaymentType PaymentType { get; set; }

        public DateTime OrderDate { get; set; }

        public long GrandTotal { get; set; }

        public OrderStatus OrderStatus { get; set; }
    }
}
